ui/apps/platform/src/Containers/Collections/CollectionFormDrawer.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { ReactElement, useEffect } from 'react';
+import {
+    Drawer,
+    DrawerActions,
+    DrawerCloseButton,
+    DrawerContent,
+    DrawerContentBody,
+    DrawerHead,
+    DrawerPanelBody,
+    DrawerPanelContent,
+    Text,
+    Title,
+} from '@patternfly/react-core';
+
+import { CollectionResponse } from 'services/CollectionsService';
+import { CollectionPageAction } from './collections.utils';
+import CollectionResults from './CollectionResults';
+import { Collection } from './types';
+import { parseCollection } from './converter';
+import CollectionForm from './CollectionForm';
+
+export type CollectionFormDrawerProps = {
+    hasWriteAccessForCollections: boolean;
+    /* The user's workflow action for this collection */
+    action: CollectionPageAction;
+    collectionData: {
+        collection: Omit<CollectionResponse, 'id'>;
+        embeddedCollections: CollectionResponse[];
+    };
+    /* Whether or not to display the collection results in an inline drawer. If false, will
+    display collection results in an overlay drawer. */
+    useInlineDrawer: boolean;
+    isDrawerOpen: boolean;
+    toggleDrawer: (isOpen: boolean) => void;
+    headerContent?: ReactElement;
+    onSubmit: (collection: Collection) => Promise<void>;
+    /* Callback used when clicking on a collection name in the CollectionAttacher section. If
+    left undefined, collection names will not be linked. */
+    appendTableLinkAction?: (collectionId: string) => void;
+};
+
+function CollectionFormDrawer({
+    hasWriteAccessForCollections,
+    action,
+    collectionData,
+    headerContent,
+    useInlineDrawer,
+    isDrawerOpen,
+    toggleDrawer,
+    onSubmit,
+}: CollectionFormDrawerProps) {
+    const initialData = parseCollection(collectionData.collection);
+    const initialEmbeddedCollections = collectionData.embeddedCollections;
+
+    useEffect(() => {
+        toggleDrawer(useInlineDrawer);
+    }, [toggleDrawer, useInlineDrawer]);
+
+    return (
+        <>
+            <Drawer isExpanded={isDrawerOpen} isInline={useInlineDrawer}>
```

Comment:
Seeing both boolean props raises pro and con of `use` versus `is` for prefix of second state variable.

Reply:
Fair point. A `use` prefix is especially bad here given the association with React.

Reply:
Ha ha, I missed that connotation.